internal/git/git.go

```diff
@@ -303,36 +364,29 @@ func (rs *RepoSource) loadRepo(path string, rg *git.Repository) (*Repo, error) {
 	r := &Repo{
 		path:       path,
 		repository: rg,
-		patch:      make(map[plumbing.Hash]*object.Patch),
 	}
-	r.commits = make(map[plumbing.Hash]*object.Commit)
-	r.trees = make(map[plumbing.Hash]*object.Tree)
-	r.refCommits = make(map[plumbing.Hash]gitypes.Commits)
+	if rs.CacheSize > 0 {
+		r.commits = lru.New(rs.CacheSize)
```

Comment:
One thing to note here: since we create multiple `lru` caches with the same cache size, the configured `CacheSize` value wouldn't reflect the total amount of items cached, instead it's a per-type setting. That's not necessarily wrong, but may be something we need to communicate in the docs.